stripe/util.py

```diff
@@ -301,9 +302,36 @@ def __get__(self, obj=None, objtype=None):
         @functools.wraps(self.method)
         def _wrapper(*args, **kwargs):
             if obj is not None:
+                self._deprecation_warning(obj)
                 return self.method(obj, *args, **kwargs)
             else:
                 class_method = getattr(objtype, self.class_method_name)
                 return class_method(*args, **kwargs)
 
         return _wrapper
+
+    def _deprecation_warning(self, obj):
+        if (
+            not isinstance(obj, stripe.api_resources.abstract.APIResource)
+            or "id" not in obj
+        ):
+            return
+
+        message = "This instance method is deprecated. Use the class method {module_name}.{class_name}.{method_name}({sid}, ...) instead.".format(
+            module_name=self._public_module_name(obj.__class__.__module__),
+            class_name=obj.__class__.__name__,
+            method_name=self.method.__name__,
+            sid=repr(obj["id"]),
+        )
+        warnings.simplefilter("always", DeprecationWarning)
+        warnings.warn(message, category=DeprecationWarning, stacklevel=3)
+        warnings.simplefilter("default", DeprecationWarning)
+
+    @staticmethod
+    def _public_module_name(module_name):
```

Comment:
Nit, but mind just adding a basic example of the before and after strings that this converts from and to in a comment? e.g. `converts "stripe.api_resources.customer" to "stripe"` or whatever it is.